client/src/components/reviews/review_form/06_UploadPhotos.jsx

```diff
@@ -1,10 +1,54 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import Requester from '../../../Requester.js';
+
+const requester = Requester();
 
 const UploadPhotos = (props) => {
+
+  const [photos, setPhotos] = useState([]);
+  const [photoUrl, setPhotoUrl] = useState([]);
```

Comment:
`ln 8` is the array to store the data for the thumbnails as I do not need to send this data anywhere, and it will be only used in this component.

`ln 9` is the array to store the transformed photo URL after uploading the image to the API endpoint. 

The `useEffect` will check if there are new photo urls added, and then add the url to the overall form object to be submitted. Because of the way I handled the form object update in the ReviewForm component, it was difficult for me to find out already existing photo urls as to not change them. Thus, I created a state in this component to make it easy to update my overall form object with the photo data required to submit a review. 

Another reason I did this is to retrieve the new url upon image upload, as opposed to waiting for the form to submit to retrieve the photo URLs, wait for the response to come back, then submit the review. This way I can separate out the order of asynchronous actions.

Reply:
Makes sense to me, thanks.